### DISCUSSION TITLE
Does $metrics typecheck when using dynamic client extensions?

### DISCUSSION THREAD

**Discussion by vedantroy** [prisma/prisma#21123] [Q&A]
### Question

I've enabled the metrics generator, and it works, but it doesn't typecheck, (VsCode is saying `$metrics` does not exist).
I'm using the dynamic client extensions feature, so this is the result of using `prisma.$extends(...)`

However, the `$metrics` object does exist directly on the `PrismaClient` object.

### Expected behavior (optional)

I should see the `$metrics` -- object on the dynamic client extension.

### Information about Prisma Schema, Client Queries and Environment (optional)

```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["metrics"]
}
```

**Comment by ludralph** [comment]
Hi @vedantroy 👋 

Thank you for raising this question.

`$metrics` at this current moment is available on the Prisma Client object but not on the extended Prisma Client.

Can you please elaborate on why you would need `$metrics` on the extended Prisma Client?

You can also open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for adding $`metrics` on the extended Prisma Client.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
